app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -184,18 +186,35 @@ class CheckAgencyController @Inject()(
         }
       }
 
-    agentSubscriptionConnector.getRegistration(knownFacts.utr, knownFacts.postcode) flatMap {
-      case Some(Registration(Some(taxpayerName), isSubscribedToAgentServices)) if !isSubscribedToAgentServices =>
-        processCheckAgencyStatus(knownFacts.utr, taxpayerName, isSubscribedToAgentServices)
-      case Some(Registration(_, isSubscribedToAgentServices)) if isSubscribedToAgentServices =>
+    subscriptionService.getSubscriptionStatus(knownFacts.utr, knownFacts.postcode).flatMap {
+      case SubscriptionProcess(SubscriptionState.BrandNewSubscription, Some(registrationDetails)) =>
+        processCheckAgencyStatus(
+          knownFacts.utr,
+          registrationDetails.taxpayerName.get,
+          registrationDetails.isSubscribedToAgentServices)
+      case SubscriptionProcess(SubscriptionState.IsOnlySubscribedInETMP, Some(registrationDetails)) =>
+        agent match {
+          case hasNonEmptyEnrolments(_) =>
+            Future successful Redirect(routes.CheckAgencyController.showHasOtherEnrolments())
```

Comment:
Do we need this redirect here? I cant see this requirement in the ticket, neither this has been on the previous version..

Reply:
https://jira.tools.tax.service.gov.uk/secure/attachment/138239/Subscription%20Journey%20-%20APB-2582.pdf

this is the graph attached with the ticket, that was updated by John

Reply:
I've just added case this as an acceptance critieria to the ticket.